libraries/AP_Motors/AP_MotorsHeli_Single.cpp

```diff
@@ -490,54 +499,61 @@ void AP_MotorsHeli_Single::output_to_motors()
         thr_lin.update_lift_max_from_batt_voltage();
     }
 
+    // Ensure we keep ddfp range and trim up to date as it is used in thrust linearisation calcs
+    bool ddfp_configured = update_ddfp_range_and_trim();
+
     switch (_spool_state) {
         case SpoolState::SHUT_DOWN:
             // sends minimum values out to the motors
             update_motor_control(ROTOR_CONTROL_STOP);
             if (_tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CW || _tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CCW){
-                rc_write(AP_MOTORS_MOT_4, calculate_ddfp_output(0.0f));
+                uint16_t ddfp_output = 0;
+                if (ddfp_configured) {
+                    ddfp_output = calculate_ddfp_output(0.0f);
+                }
+                rc_write(AP_MOTORS_MOT_4, ddfp_output);
             }
             break;
         case SpoolState::GROUND_IDLE:
             // sends idle output to motors when armed. rotor could be static or turning (autorotation)
             update_motor_control(ROTOR_CONTROL_IDLE);
             if (_tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CW || _tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CCW){
-                rc_write(AP_MOTORS_MOT_4, calculate_ddfp_output(0.0f));
+                uint16_t ddfp_output = 0;
+                if (ddfp_configured) {
+                    ddfp_output = calculate_ddfp_output(0.0f);
+                }
+                rc_write(AP_MOTORS_MOT_4, ddfp_output);
             }
             break;
         case SpoolState::SPOOLING_UP:
         case SpoolState::THROTTLE_UNLIMITED:
             // set motor output based on thrust requests
             update_motor_control(ROTOR_CONTROL_ACTIVE);
             if (_tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CW || _tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CCW){
-                SRV_Channel *c = SRV_Channels::srv_channel(AP_MOTORS_MOT_4);
-                if (c != nullptr) {
-                    _ddfp_pwm_min = c->get_output_min();
-                    _ddfp_pwm_max = c->get_output_max();
-                    _ddfp_pwm_trim = c->get_trim();
-                }
                 float servo_out = 0.0f;
-                if (is_positive((float) (_ddfp_pwm_max - _ddfp_pwm_min))) {
+                if (ddfp_configured) {
                     float servo4_trim = constrain_float((_ddfp_pwm_trim - 1000) / (_ddfp_pwm_max - _ddfp_pwm_min), 0.0f, 1.0f);
```

Comment:
```suggestion
                    float servo4_trim = constrain_float((_ddfp_pwm_trim - _ddfp_pwm_min) / (_ddfp_pwm_max - _ddfp_pwm_min), 0.0f, 1.0f);
```
The 1000 here meas were more or less ignoring the PWM min.